README.md

```diff
@@ -917,6 +915,30 @@ job.depends([job1, job2]);
 // Clear any dependencies previously added on this job
 job.depends();
 ```
+Added antecendent jobs must have already had `.save()` run on them, so they will have the `_id` attribute that
+is used to form the dependency. In practice, that often means that the dependent job will be created within the `save()` callback of its antecedent:
+
+```javascript
+// Create a job:
+var sendJob = new Job(myJobs, 'sendEmail', {
+  address: 'bozo@clowns.com',
+  subject: 'Critical rainbow hair shortage',
+  message: 'LOL; JK, KThxBye.'
+})
+.save(function(err, jobId) {
+  if (err) {
+    console.log("Couldn't create sendEmail job!");
+    return false;
+  }
+  var archiveJob = new Job(myJobs, 'archiveEmail', {
+    emailData: sendJob.data
+  }).depends([sendJob]).save();
+});
+```
+Note, however, that you *cannot* use `depends()` to reliably pass the `result` from an antecedent `job1` to a dependent `job2` as data. That is because `job2`, including its data object, will be created (though not run!) right away when the `new` keyword is encountered. At that point, `job1` may not yet be done, and its `result` nonexistent. It is therefore best to combine multiple jobs into one if they rely on one another's result. If that is not possible, because the jobs are to be separated in time (i.e. delayed) or space (i.e. run on separate machines), then two options are:
+
```

Comment:
I have mixed feelings about this... I really appreciate you taking the time to put this PR together, and I understand that this is what tripped you up, but this is a lot of details (in an already too long document). It almost feels like what is necessary here is a clarification in `job.save()` that it returns (with Fibers) or invokes its callback (when provided) immediately after the job is saved in the database, and not when the job finishes running. That wouldn't even be possible because the server could restart between invoking `job.save()` and when the job actually runs...